### DISCUSSION TITLE
Gauge API time out error

### DISCUSSION THREAD

**Discussion by dibyaranjan-pg** [getgauge/gauge#2032] [General]
Suddenly we started getting api time out error in Gauge while running our tests. 

Gauge Version: 1.2.1
Gauge Java: 0.7.15 (tried with last 2 version as well)

How do we run: 
We create a docker image with all the possible pre-req and run the docker image in our build machine (Linux). We have 20 variants of tests and run them in parallel. each variant gets its own configuration and runs in a docker container. From last 2/3 days, constantly geting gauge java api time out error. for some variants. Some variants works fine.

`[[1;36mDEBUG[m]   (f) dir = /app
[[1;36mDEBUG[m]   (f) flags = [--env=integration]
[[1;36mDEBUG[m]   (f) inParallel = true
[[1;36mDEBUG[m]   (f) nodes = 10
[[1;36mDEBUG[m]   (f) skip = false
[[1;36mDEBUG[m]   (f) skipTests = false
[[1;36mDEBUG[m]   (f) specsDir = specs/
[[1;36mDEBUG[m]   (f) tags = REGION=DD&SERVICE=guruland-guruland&TYPE=api&,\!in-progress,\!EXCLUDE=integration
[[1;36mDEBUG[m] -- end configuration --
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by com.google.inject.internal.cglib.core.$ReflectUtils$1 (file:/usr/share/maven/lib/guice.jar) to method java.lang.ClassLoader.defineClass(java.lang.String,byte[],int,int,java.security.ProtectionDomain)
WARNING: Please consider reporting this to the maintainers of com.google.inject.internal.cglib.core.$ReflectUtils$1
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
Error ----------------------------------

[Gauge]
Failed to start gauge API: Timed out connecting to java

Get Support ----------------------------
	Docs:          https://docs.gauge.org
	Bugs:          https://github.com/getgauge/gauge/issues
	Chat:          https://github.com/getgauge/gauge/discussions

Your Environment Information -----------
	linux, 1.1.7, 5d86b72
	html-report (4.0.12), java (0.7.13), json-report (0.3.6), screenshot (0.0.1)
[[1;34mINFO[m] [1m------------------------------------------------------------------------[m
[[1;34mINFO[m] [1;31mBUILD FAILURE[m
[[1;34mINFO[m] [1m------------------------------------------------------------------------[m
[[1;34mINFO[m] Total time:  01:07 min
[[1;34mINFO[m] Finished at: 2021-06-29T20:31:31+08:00
[[1;34mINFO[m] [1m------------------------------------------------------------------------[m
[[1;31mERROR[m] Failed to execute goal [32mcom.thoughtworks.gauge.maven:gauge-maven-plugin:1.4.3:execute[m [1m(default-cli)[m on project [36mgurutest-gauge[m: [1;31mGauge Specs execution failed. null[m: GaugeExecutionFailedException -> [1m[Help 1][m
[1;31morg.apache.maven.lifecycle.LifecycleExecutionException[m: [1;31mFailed to execute goal [32mcom.thoughtworks.gauge.maven:gauge-maven-plugin:1.4.3:execute[m [1m(default-cli)[m on project [36mgurutest-gauge[m: [1;31mGauge Specs execution failed. null[m[m
    [1mat[m org.apache.maven.lifecycle.internal.MojoExecutor.execute ([1mMojoExecutor.java:215[m)
    [1mat[m org.apache.maven.lifecycle.internal.MojoExecutor.execute ([1mMojoExecutor.java:156[m)
    [1mat[m org.apache.maven.lifecycle.internal.MojoExecutor.execute ([1mMojoExecutor.java:148[m)
    [1mat[m org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject ([1mLifecycleModuleBuilder.java:117[m)
    [1mat[m org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject ([1mLifecycleModuleBuilder.java:81[m)
    [1mat[m org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build ([1mSingleThreadedBuilder.java:56[m)
    [1mat[m org.apache.maven.lifecycle.internal.LifecycleStarter.execute ([1mLifecycleStarter.java:128[m)`

**Comment by sriv** [comment]
What does `gauge.log` state? You may find some hints there. If you can share an image/project for me to replicate, I can help investigate.

**Reply by dibyaranjan-pg** [reply]
@sriv I could fix this by increasing the runner timeout which is present under `.gauge/config` folder. Previously was changing the values under project root folder.
